src/coprocessor/tracker.rs

```diff
@@ -0,0 +1,274 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use kvproto::kvrpcpb;
+
+use storage::engine::{PerfStatisticsDelta, PerfStatisticsInstant};
+use util::futurepool;
+use util::time::{self, Duration, Instant};
+
+use coprocessor::dag::executor::ExecutorMetrics;
+use coprocessor::*;
+
+// If handle time is larger than the lower bound, the query is considered as slow query.
+const SLOW_QUERY_LOWER_BOUND: f64 = 1.0; // 1 second.
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+enum TrackerState {
```

Comment:
can you split your PR? Your PR is too huge.

Maybe you can move tracker to another PR at first.

Reply:
Ok